src/index/hnsw/hnsw_config.h

```diff
@@ -15,11 +15,15 @@
 #include "knowhere/comp/index_param.h"
 #include "knowhere/config.h"
 
-// This is not a valid EF value for HNSW
-// This value is used to tell if HnswConfig.ef is coming from user or not
-const int64_t kDefaultHnswEfPlaceholder = -1;
-
 namespace knowhere {
+
+namespace {
+
+constexpr const CFG_INT::value_type kDefaultEfDivider = 16;
```

Comment:
what "Divider" mean ?

Reply:
meaning if k < divider, ef = divider. Anyway, more than welcome to better name, I agree it is poor